File: src/main/scala/performance/Benchmark.scala
package performance

import org.apache.spark.sql.SparkSession

object Benchmark {

  def run(f: => Unit): Unit = {
    val startTime = System.nanoTime
    f
    val endTime = System.nanoTime
    println(s"Time taken ${(endTime - startTime).toDouble / 1000000000} seconds")
  }

  def withWarmUp(spark: SparkSession)(f: => Unit): Unit = {
    spark.range(1000L * 1000 * 1000).selectExpr("sum(id)").collect()
    run(f)
  }
}

File: src/main/java/javachess/performance/Benchmark.java
package javachess.performance;

import org.apache.spark.sql.SparkSession;

public class Benchmark {

    public static void run(Runnable f) {
        long startTime = System.nanoTime();
        f.run();
        long endTime = System.nanoTime();
        System.out.println("Time taken " + (endTime - startTime) / 1000_000_000.0 + " seconds");
    }

    public static void withWarmUp(SparkSession spark, Runnable f) {
        spark.range(1000L * 1000 * 1000).selectExpr("sum(id)").collect();
        run(f);
    }
}
